src/tribler/core/components/bandwidth_accounting/tests/test_community.py

```diff
@@ -28,128 +30,121 @@ def create_node(self):
 
-        from_pk = self.nodes[0].my_peer.public_key.key_to_bin()
-        assert self.nodes[0].overlay.database.get_total_taken(from_pk) == 2000
-        assert self.nodes[1].overlay.database.get_total_taken(from_pk) == 2000
+        assert self.database(ID1).get_total_taken(self.key_bin(ID1)) == 2000
+        assert self.database(ID2).get_total_taken(self.key_bin(ID1)) == 2000
 
     async def test_bilateral_transaction(self):
         """
         Test creating a transaction from A to B and one from B to A.
         """
-        await self.nodes[0].overlay.do_payout(self.nodes[1].overlay.my_peer, 500)
-        await self.nodes[1].overlay.do_payout(self.nodes[0].overlay.my_peer, 1500)
+        await self.overlay(ID1).do_payout(self.peer(ID2), 500)
+        await self.overlay(ID2).do_payout(self.peer(ID1), 1500)
 
-        pk1 = self.nodes[0].my_peer.public_key.key_to_bin()
-        pk2 = self.nodes[1].my_peer.public_key.key_to_bin()
-        assert self.nodes[0].overlay.database.get_total_taken(pk1) == 500
-        assert self.nodes[1].overlay.database.get_total_taken(pk1) == 500
-        assert self.nodes[0].overlay.database.get_total_taken(pk2) == 1500
-        assert self.nodes[1].overlay.database.get_total_taken(pk2) == 1500
+        assert self.database(ID1).get_total_taken(self.key_bin(ID1)) == 500
+        assert self.database(ID2).get_total_taken(self.key_bin(ID1)) == 500
+        assert self.database(ID1).get_total_taken(self.key_bin(ID2)) == 1500
+        assert self.database(ID2).get_total_taken(self.key_bin(ID2)) == 1500
 
     async def test_bilateral_transaction_timestamps(self):
         """
         Test creating subsequent transactions and check whether the timestamps are different.
         """
-        tx1 = await self.nodes[0].overlay.do_payout(self.nodes[1].overlay.my_peer, 500)
-        await sleep(0.1)
-        tx2 = await self.nodes[0].overlay.do_payout(self.nodes[1].overlay.my_peer, 500)
+        tx1 = await self.overlay(ID1).do_payout(self.peer(ID2), 500)
+        tx2 = await self.overlay(ID1).do_payout(self.peer(ID2), 500)
+
         assert tx1.timestamp != tx2.timestamp
 
     async def test_invalid_transaction(self):
         """
         Test sending a transaction with an invalid signature to the counterparty, which should be ignored.
         """
-        other_peer = self.nodes[1].my_peer
-        tx = self.nodes[0].overlay.construct_signed_transaction(other_peer, 300)
+        tx = self.overlay(ID1).construct_signed_transaction(self.peer(ID2), 300)
         tx.signature_a = b"invalid"
-        self.nodes[0].overlay.database.BandwidthTransaction.insert(tx)
-        cache = self.nodes[0].overlay.request_cache.add(BandwidthTransactionSignCache(self.nodes[0].overlay, tx))
-        self.nodes[0].overlay.send_transaction(tx, other_peer.address, cache.number)
+        self.database(ID1).BandwidthTransaction.insert(tx)
+        cache = self.add_cache(ID1, BandwidthTransactionSignCache(self.overlay(0), tx))
```

Comment:
```suggestion
        cache = self.add_cache(ID1, BandwidthTransactionSignCache(self.overlay(ID1), tx))
```

I forgot to convert the node id 0 into its corresponding variable.